src/Shared/OpportunisticIntern.cs

```diff
@@ -752,147 +763,56 @@ internal void ReportStatistics(string heading)
                     }
                     else if (length == 4)
                     {
-                        if (candidate[0] == 'T')
-                        {
-                            if (candidate[1] == 'R' && candidate[2] == 'U' && candidate[3] == 'E')
-                            {
-                                interned = "TRUE";
-                                return true;
-                            }
-
-                            if (candidate[1] == 'r' && candidate[2] == 'u' && candidate[3] == 'e')
-                            {
-                                interned = "True";
-                                return true;
-                            }
-                        }
-
-                        if (candidate[0] == 'C' && candidate[1] == 'o' && candidate[2] == 'p' && candidate[3] == 'y')
-                        {
-                            interned = "Copy";
-                            return true;
-                        }
-
-                        if (candidate[0] == 't' && candidate[1] == 'r' && candidate[2] == 'u' && candidate[3] == 'e')
-                        {
-                            interned = "true";
-                            return true;
-                        }
-
-                        if (candidate[0] == 'v' && candidate[1] == '4' && candidate[2] == '.' && candidate[3] == '0')
+                        if (TryInternHardcodedString(candidate, "TRUE", ref interned) ||
+                            TryInternHardcodedString(candidate, "True", ref interned) ||
+                            TryInternHardcodedString(candidate, "Copy", ref interned) ||
+                            TryInternHardcodedString(candidate, "true", ref interned) ||
+                            TryInternHardcodedString(candidate, "v4.0", ref interned))
```

Comment:
Amazing how much clearer this is. And it makes it obvious that we should think about adding some new common strings like `netstandard2.0` and `net48` and friends.

Reply:
Yup, new strings are coming soon.